core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveFloatingCodec.java

```diff
@@ -139,15 +139,15 @@ public void encode(int rowId, long value) {
     @Override
     public void encode(int rowId, float value) {
       if (targetDataType == DataTypes.BYTE) {
-        encodedPage.putByte(rowId, (byte) (value * factor));
+        encodedPage.putByte(rowId, (byte) (value * (float) factor));
```

Comment:
don't typecast it every time, take another float variable at class level

Reply:
added another variable for float